c/passTicketService.c

```diff
@@ -0,0 +1,271 @@
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+*/
+
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <stdint.h>
+
+#include "zowetypes.h"
+#include "alloc.h"
+#include "utils.h"
+#include "charsets.h"
+#include "httpserver.h"
+#include "json.h"
+#include "zssLogging.h"
+#include "passTicketService.h"
+#include "serviceUtils.h"
+
+#define TICKET_LEN 8
+#define APP_NAME_KEY "applicationName"
+#define APP_NAME_KEY_LEN 8
```

Comment:
Is there a reason we are limiting app name to 8? I don't see any such restriction in:

https://docs.zowe.org/stable/extend/extend-apiml/api-mediation-passtickets/

Reply:
Yes, application name is only 1-8 byte long: https://www.ibm.com/docs/en/zos/2.4.0?topic=3-generate-passticket1-4

Reply:
Nevermind, just found elsewhere "The APPLID is the name of the application (up to 8 characters) that is used by security products to differentiate certain security operations (like PassTickets) between applications."